### PR TITLE
fix: always set exit code if exiting uncleanly

### LINKED ISSUE #7672: [BUG] npm error Exit handler never called exits with status code 0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

This issue, related to #7666 and #7669, exits with status code 0 when the exit handler npm error occurs. Exiting with status code 0 allows the ci to continue to the next steps.

### Expected Behavior

I would expect it to exit with status code 1 since it is an error. That way ci would fail when this error occurs.

### Steps To Reproduce

The GitHub Actions workflow showing this behaviour can be found at: https://github.com/Sebastian-Webster/npm-reproducible-example/actions/runs/10007253675
The repository code for that workflow can be found here at: https://github.com/Sebastian-Webster/npm-reproducible-example/tree/22.5.0
This example shows that the issue does not happen in Node 22.4.1, but does in 22.5.0.

### Environment

- npm: v10.8.2
- Node.js: v22.5.0
- OS Name: Windows 2019 & 2022, macOS 12, 13, & 14, and Ubuntu 20.04 & Ubuntu 22.04
- System Model Name: GitHub Actions

### PR DESCRIPTION
**Pull request by wraithgar**
An erroneous assumption was that if this was explicitly set, then the exit was still intentional.  This is not the case.

Closes: https://github.com/npm/cli/issues/7672

### CODE REVIEW

File: lib/cli/exit-handler.js [hunk 1]
```diff
@@ -66,8 +66,9 @@ class ExitHandler {
   }
 
   #handleProcessExit (code) {
-    // Force exit code to a number if it has not been set
-    const exitCode = typeof code === 'number' ? code : (this.#exited ? 0 : 1)
+    const numCode = Number(code) || 0
+    // Always exit w/ a code if exit handler was not called
```
Comment (hashtagchris) [inline lib/cli/exit-handler.js]:
```suggestion
    // Always exit w/ a non-zero code if exit handler was not called
```

File: test/lib/cli/exit-handler.js [hunk 2]
```diff
@@ -194,15 +194,16 @@ t.test('exit handler never called', async t => {
     const { logs, errors } = await mockExitHandler(t, {
       config: { loglevel: 'silent' },
     })
-    process.emit('exit', 1)
+    process.emit('exit', 0)
     t.strictSame(logs, [])
     t.strictSame(errors(), [''], 'one empty string')
+    t.equal(process.exitCode, 1)
```
Comment (hashtagchris) [inline test/lib/cli/exit-handler.js:200]: Optional: Add a dedicated test that verifies 0 is replaced by a non-zero value. The test name might make things click for someone.

**Review by hashtagchris — APPROVED** [review]
Looks good. Left optional suggestions.
